### PR TITLE
Add `dpnp.linalg.tensorinv()` implementation

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
This PR adds a new `dpnp.linalg.tensorinv()` function to calculate the inverse of input tensor arrray using `dpnp.linalg.inv()`.
The changes are related to support of all types.

- [X] Have you provided a meaningful PR description?
- [X] Have you added a test, reproducer or referred to issue with a reproducer?
- [X] Have you tested your changes locally for CPU and GPU devices?
- [X] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/linalg/dpnp_iface_linalg.py [hunk 1]
```diff
@@ -897,3 +898,61 @@ def slogdet(a):
     check_stacked_square(a)
 
     return dpnp_slogdet(a)
+
+
+def tensorinv(a, ind=2):
+    """
+    Compute the `inverse` of a tensor.
```
Comment (antonwolfy) [inline dpnp/linalg/dpnp_iface_linalg.py:905]:
```suggestion
    Compute the 'inverse' of an N-dimensional array.
```

File: dpnp/linalg/dpnp_iface_linalg.py [hunk 2]
```diff
@@ -897,3 +898,61 @@ def slogdet(a):
...
+
+    For full documentation refer to :obj:`numpy.linalg.tensorinv`.
+
+    Parameters
+    ----------
+    a : {dpnp.ndarray, usm_ndarray}
+        Tensor to `invert`. Its shape must be 'square', i. e.,
+        ``prod(a.shape[:ind]) == prod(a.shape[ind:])``.
+    ind : int
```
Comment (antonwolfy) [inline dpnp/linalg/dpnp_iface_linalg.py:914]:
```suggestion
    ind : int, optional
```

**Review by antonwolfy — APPROVED** [review]
Thank you @vlad-perevezentsev.
I only have a couple of minor comments.

File: dpnp/linalg/dpnp_iface_linalg.py (hunk 1 above)
Comment (vlad-perevezentsev) [reply dpnp/linalg/dpnp_iface_linalg.py:905]: I will fix it in #1753 
Thank you

File: dpnp/linalg/dpnp_iface_linalg.py (hunk 2 above)
Comment (vlad-perevezentsev) [reply dpnp/linalg/dpnp_iface_linalg.py:914]: same

**Pull request merged**